tokio-util/Cargo.toml

```diff
@@ -56,6 +58,7 @@ async-stream = "0.3.0"
 futures = "0.3.0"
 futures-test = "0.3.5"
 parking_lot = "0.12.0"
+tempfile = "3.1.0"
```

Comment:
Adding a dependency to add a test feels overkill. 
But the thing is, the test only fails for a `tokio::fs::file::File`.
Using something like `tokio::io::util::buf_writer::BufWriter` or `std::io::cursor::Cursor` seem to be working fine on latest `upstream/master`:
```rust
#[tokio::test]
async fn compat_seek() -> futures_util::io::Result<()> {
    let mut cursor = Cursor::new(Vec::new()).compat_write();

    cursor.write_all(&[0, 1, 2, 3, 4, 5]).await?;
    cursor.write_all(&[6, 7]).await?;

    assert_eq!(cursor.stream_position().await?, 8);

    // Modify elements at position 2.
    assert_eq!(cursor.seek(SeekFrom::Start(2)).await.unwrap(), 2);
    cursor.write_all(&[8, 9]).await.unwrap();

    cursor.flush().await.unwrap();

    // Verify we still have 8 elements
    assert_eq!(cursor.seek(SeekFrom::End(0)).await?, 8);
    // Seek back to the start of the file to read and verify contents.
    cursor.seek(SeekFrom::Start(0)).await?;

    let mut buf = Vec::new();
    let num_bytes = cursor.read_to_end(&mut buf).await?;
    assert_eq!(&buf[..num_bytes], &[0, 1, 8, 9, 4, 5, 6, 7]);

    Ok(())
}
```

Reply:
@Darksonn, is there some alternative way to test this? And would like more sense to skip adding one if there is none?

Reply:
This dependency already exists in the main Tokio crate, so adding it here doesn't _actually_ add a new dependency.